src/Abp/Runtime/Caching/AbpCacheBase.cs

```diff
@@ -32,104 +33,87 @@ protected AbpCacheBase(string name) : base(name)
 
         public virtual TValue Get(TKey key, Func<TKey, TValue> factory)
         {
-            TValue item = default(TValue);
 
-            try
+            if (TryGetValue(key, out TValue value))
             {
-                item = GetOrDefault(key);
-            }
-            catch (Exception ex)
-            {
-                Logger.Error(ex.ToString(), ex);
+                return value;
             }
 
-            if (item == null)
+            lock (SyncObj)
             {
-                lock (SyncObj)
+                if (TryGetValue(key, out value))
+                {
+                    return value;
+                }
+
+                var generatedValue = factory(key);
+                if (!EqualityComparer<TValue>.Default.Equals(generatedValue, default))
                 {
                     try
                     {
-                        item = GetOrDefault(key);
+                        Set(key, generatedValue);
                     }
                     catch (Exception ex)
                     {
                         Logger.Error(ex.ToString(), ex);
                     }
-
-                    if (item == null)
-                    {
-                        item = factory(key);
-
-                        if (item == null)
-                        {
-                            return default(TValue);
-                        }
-
-                        try
-                        {
-                            Set(key, item);
-                        }
-                        catch (Exception ex)
-                        {
-                            Logger.Error(ex.ToString(), ex);
-                        }
-                    }
                 }
+                return generatedValue;
```

Comment:
Why change to `return` inside the lock?

Reply:
As I removed `TValue item = default(TValue);`, we do not need to store the result to return at the end of the function.

Returning within a using lock is fine as the using scope will be disposed eventually (we also returning result within using block in many places).
See https://stackoverflow.com/questions/11776945/is-it-a-good-approach-to-call-return-inside-using-statement/11776998